src/test/java/DereferenceTesting.java

```diff
@@ -0,0 +1,167 @@
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Class to perform JUnit tests on the AddNullCheckBeforeDereferenceRefactoring
+ * refactoring module
+ * 
+ * @see DereferenceTesting
+ */
+public class DereferenceTesting {
+
+	public void test(String input, String expectedOutput) {
+		TestingEngine.testSingleRefactoring(input, expectedOutput, AddNullCheckBeforeDereferenceRefactoring.NAME);
+	}
+
+	@Test
+	public void simpleTest() {
+		String input = """
+				public class Test {
+					private void test() {
+						Class<?> dependentObj = (independentObj != null ? independentObj.getDependent() : null);
+						if (dependentObj != null) {
+							;
+						}
+					}
+				}
+				""";
+		String expectedOutput = """
+				public class Test {
+					private void test() {
+						Class<?> dependentObj = (independentObj != null ? independentObj.getDependent() : null);
+						if ((independentObj != null ? independentObj.getDependent() : null)) {
```

Comment:
This doesn't look right: I'd expect the guard in the output to just be `independentObj != null` rather than the whole expression.

Reply:
This problem is pervasive in all of these new tests that use conditional expressions. In these cases, I'd expect that we want to replace the test of `dependentObj` to be a test of `independentObj` (on which `dependentObj`'s value depends). If we leave this as it is, there's a danger of creating side-effects: we'll be calling whatever the non-null side of the conditional is twice instead of once. That code could do something arbitrary (e.g., remove something from a list or queue to get a value?) that shouldn't be repeated, so it's critical that we avoid doing it an extra time.